src/external/spglib.jl

```diff
@@ -88,26 +94,47 @@ spglib_cell(lattice, atoms) = first(spglib_cell_atommapping_(lattice, atoms))
     Stildes, τtildes
 end
 
-function spglib_standardize_cell(lattice::MatT, atoms; correct_symmetry=true,
-                                 primitive=false, tol_symmetry=1e-5) where {MatT}
-    spglib = pyimport("spglib")
-    T = eltype(lattice)
-
+function spglib_standardize_cell(lattice::AbstractArray{T}, atoms; correct_symmetry=true,
+                                 primitive=false, tol_symmetry=1e-5) where {T}
+                                 
     # Convert lattice and atoms to spglib and keep the mapping between our atoms
+    spg_lattice = Matrix{Float64}(lattice) 
     # and spglibs atoms
-    cell, atommapping = spglib_cell_atommapping_(lattice, atoms)
+    spg_atoms, atommapping = spglib_atommapping(atoms)
+    spg_positions, spg_numbers = spg_atoms
 
     # Ask spglib to standardize the cell (i.e. find a cell, which fits the spglib conventions)
-    res = spglib.standardize_cell(spglib_cell(lattice, atoms), to_primitive=primitive,
-                                  no_idealize=!correct_symmetry, symprec=tol_symmetry)
-    spg_lattice, spg_scaled_positions, spg_numbers = res
+    num_atoms = ccall((:spg_standardize_cell, SPGLIB), Cint,
+      (Ptr{Cdouble}, Ptr{Cdouble}, Ptr{Cint}, Cint, Cint, Cint, Cdouble),
+      spg_lattice, spg_positions, spg_numbers, length(spg_numbers), Cint(primitive),
+      Cint(!correct_symmetry), tol_symmetry)
 
     # Note: In the python interface of spglib the lattice vectors
     #       are given in rows, but DFTK uses columns
     #       For future reference: The C interface spglib also uses columns.
-    newlattice = MatT(spg_lattice')
     newatoms = [(atommapping[iatom]
-                 => T.(spg_scaled_positions[findall(isequal(iatom), spg_numbers), :]))
+                 => T.(spg_positions[findall(isequal(iatom), spg_numbers), :]))
                 for iatom in unique(spg_numbers)]
-    newlattice, newatoms
+    spg_lattice, newatoms
+end
+
+function spglib_get_stabilized_reciprocal_mesh(kgrid_size, rotations;
+                                               is_shift=Vec3(0, 0, 0),
+                                               is_time_reversal=false,
+                                               qpoints=[Vec3(0.0, 0.0, 0.0)],
+                                               isdense=false)
+
+    rotations = eltype(rotations) == Int32 ? rotations : Cint.(rotations)
+    nkpt = prod(kgrid_size)
+    mapping = Vector{Cint}(undef, nkpt)
+    grid_address = Matrix{Cint}(undef, 3, nkpt)
+    qpoints = eltype(qpoints) == Real ? [qpoints] : qpoints # Note: Done similarly to the python spglib
```

Comment:
I'd put the comment on the preceeding line. If you reference to the python implementation, could you add a link to the file on github (use a commit hash which is static and not a branch name when you deep-link)?

Reply:
I actually opted out of doing this and just forcing a Vector{Vec3{Float64}} as the type for qpoints, there are too many possible things one could throw in and catching all of them may not be easy/error prone (I'm not sure this line was actually correct).

Reply:
I think that is a good idea. But again I would not not opting for too strong type constraints and rather explicitly convert to `Vector{Vec3{Float64}}` inside the function body. Just a `Vec3{Float64}.(qpoints)` should do, I'd say.